File: launchschool_exercises/210_small_problems/easy1/sum-or-product.js
var x = parseInt(prompt('Please enter an integer greater than 0: '));
var op = prompt('Enter "s" to compute the sum, or "p" to compute the product. ');

var total = 1;
for (var i = 2; i <= x; i++) {
  if (op === 's') {
    total += i;
  } else {
    total *= i;
  }
} 

if (op === 's') {
  console.log('The sum of the integers between 1 and ' + x + 'is ' + total + '.');
} else {
    console.log('The product of the integers between 1 and ' + x + 'is ' + total + '.');
}

File: launchschool_exercises/109_small_problems/easy2/sum_or_product.rb
def compute(num, op)
  if op == 's'
    count = 1
    total = 0
    while count <= num
      total += count
      count += 1
    end
  else
    count = 1
    total = 1
    while count <= num
      total *= count
      count += 1
    end
  end
  total
end

print '>> Please enter an integer greater than 0: '
num = gets.chomp.to_i

print ">> Enter 's' to compute the sum, 'p' to compute the product: "
op = gets.chomp
raise 'Invalid input' unless op.casecmp('s') || op.casecmp('p')

result = compute(num, op)
if op == 's'
  puts "The sum of the integers between 1 and #{num} is #{result}."
elsif op == 'p'
  puts "The product of the integers between 1 and #{num} is #{result}."
end
